nncf/experimental/onnx/graph/onnx_graph.py

```diff
@@ -248,24 +249,44 @@ def get_tensor_shape(self, tensor: ValueInfoProto) -> List[int]:
 
     def get_edge_shape(self, edge_name: str) -> List[int]:
         """
-        Returns tensor shape of the edge with the name 'edge_name'.
-        """
-        for tensor in self.activations_tensors:
-            if tensor.name == edge_name:
-                return self.get_tensor_shape(tensor)
+        Returns a shape of the edge with the name 'edge_name'.
+        If the activations tensors were not filled in self._activations_tensor_name_to_value_info, it updates them.
+        If after updating of the self._activations_tensor_name_to_value_info, there is still no such tensor,
+        do shape inference of the model.
+        :param edge_name: The name of the edge.
+        :return: Shape of the tensor on that edge.
+        """
+        if self._activations_tensor_name_to_value_info is None:
+            self._update_activation_tensors()
+        if edge_name in self._activations_tensor_name_to_value_info:
+            return self.get_tensor_shape(self._activations_tensor_name_to_value_info[edge_name])
+        self._update_activation_tensors(do_shape_inference=True)
```

Comment:
Is this right? I don't think that doing shape inference at every call of `get_edge_shape` is efficient, or lazy.

Reply:
Sorrz, missed the early exit clause.